packages/mobile/src/web3/dataEncryptionKey.ts

```diff
@@ -118,23 +189,129 @@ export function* registerAccountDek(account: string) {
   }
 }
 
+// Unlike normal DEK registration, registration via Komenci should be considered fatal. If there
+// is no on-chain mapping of accountAddresss => walletAddress, then senders will erroneously
+// send to MTW instead of EOA. A no-op if registration has already been done
+export function* registerWalletAndDekViaKomenci(
+  komenciKit: KomenciKit,
+  accountAddress: string,
+  walletAddress: string
+) {
+  ValoraAnalytics.track(OnboardingEvents.account_dek_register_start)
+
+  Logger.debug(
+    `${TAG}@registerAccountDekViaKomenci`,
+    'Setting wallet address and public data encryption key'
+  )
+
+  // Don't think we need to unlock the account anymore because we do
+  // it at the start of the feeless verification flow
+  // yield call(getConnectedUnlockedAccount)
+  ValoraAnalytics.track(OnboardingEvents.account_dek_register_account_unlocked)
```

Comment:
We should ensure the account is unlocked here because if the unlock timeout has passed (e.g. the use backgrounded the app during the verification process and came back after > 10 minutes) then we either need to unlock here or bail.

Reply:
Yeah makes sense. After looking at the function, I now see that it does nothing if the account is already unlocked so I guess there is no harm in including it.